api/src/main/java/marquez/service/LineageService.java

```diff
@@ -49,14 +50,27 @@ public LineageService(LineageDao delegate, JobDao jobDao) {
 
   // TODO make input parameters easily extendable if adding more options like 'withJobFacets'
   public Lineage lineage(NodeId nodeId, int depth, boolean withRunFacets) {
+    log.debug("Attempting to get lineage for node '{}' with depth '{}'", nodeId.getValue(), depth);
     Optional<UUID> optionalUUID = getJobUuid(nodeId);
     if (optionalUUID.isEmpty()) {
-      throw new NodeIdNotFoundException("Could not find node");
+      throw new NodeIdNotFoundException(String.format("Node '%s' not found!", nodeId.getValue()));
     }
```

Comment:
Ok, minor change, but can we just update this block to work like the block you just added - if there is no job data, just return lineage for the dataset. This could be considered an API change, but it's not specified in the contract and, personally, I think it's just broken.

Reply:
Yeah, I think that's a reasonable default at least until we revisit the behavior on missing relationships.

>  I think it's just broken

Are you referring to the `LineageAPI` or a broken relationship between nodes?

Reply:
The fact that we throw an exception if we don't find a job associated with the dataset. In the code you just added, and in the PR I merged, if no job is _currently_ associated with the dataset, we return a lineage graph with a single node. But in this block, if no job was _ever_ associated with the dataset, we throw an exception. Why not just return the dataset node as we do below?

Reply:
Yeah, I was going to return a single node, but was just curious (and wanted clarification) on what you meant by "just broken" :p

Reply:
Got it - I misread your original statement to suggest that the _current_ behavior was correct 🤦🏽‍♂️
By "just broken", I mean to say it's a poor user experience. It works for jobs, it should work for datasets, too, IMO.

Reply:
Agreed! and no worries ;)